Remotion/ObjectBinding/Web/UI/Controls/BocList.RowMenuSupport.cs

```diff
@@ -80,6 +94,52 @@ private void EnsureRowMenusInitialized ()
       _rowMenus = rowMenus.ToArray();
     }
 
+    private bool TryGetRowMenu (string targetInfo, [NotNullWhen(true)] out string? id)
+    {
+      var rowMenuIDPrefix = UniqueID + c_rowMenuIDPrefix;
+      if (targetInfo.StartsWith(rowMenuIDPrefix))
+      {
+        id = targetInfo.Substring(rowMenuIDPrefix.Length);
+        return true;
+      }
+
+      id = null;
+      return false;
+    }
+
+    private void EnsureRowMenuInitialized (string id)
+    {
+      if (_rowMenusInitialized)
+        return;
+
+      _singleRowMenuInitialized = true;
+
+      if (! AreRowMenusEnabled)
+        return;
+      if (!HasValue)
+        return;
+      EnsureChildControls();
+
+      Assertion.IsTrue(_rowMenus.Length == 0);
+      Assertion.IsTrue(_rowMenusPlaceHolder.Controls.Count == 0);
+
+      var rowMenus = new List<BocListRowMenu>();
+      var row = EnsureBocListRowsForCurrentPageGot().Single(r => GetRowMenuID(r.ValueRow).Contains(id));
+
+      var rowMenu = new BocListRowMenu(this, row.ValueRow);
+      rowMenu.ID = GetRowMenuID(row.ValueRow);
+      rowMenu.EventCommandClick += (sender, e) => HandleRowMenuItemClick(sender, e, OnRowMenuItemEventCommandClick);
+      rowMenu.WxeFunctionCommandClick += (sender, e) => HandleRowMenuItemClick(sender, e, OnRowMenuItemWxeFunctionCommandClick);
+
+      _rowMenusPlaceHolder.Controls.Add(rowMenu);
+      WebMenuItem[] menuItems = InitializeRowMenuItems(row.ValueRow.BusinessObject, row.ValueRow.Index);
+      rowMenu.MenuItems.AddRange(menuItems);
+
+      rowMenus.Add(rowMenu);
```

Comment:
We are duplicating this part of the code, which makes it easy to forget changes when updating one of the places. I think we should extract this logic into its own method